.github/workflows/lint.yml

```diff
@@ -25,15 +25,13 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 black
+        pip install black ruff
 
-    - name: Lint with flake8
+    - name: Lint with black
       run: |
-        # stop the build if there are Python syntax errors or undefined names
-        flake8 . --select=E9,F63,F7,F82 --show-source
-        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-        flake8 . --exit-zero --max-complexity=5
+        black --check --skip-string-normalization --line-length 120 rq tests
 
-    - name: Lint with black
+    - name: Lint with ruff
       run: |
-        black -S -l 120 rq/
+        # stop the build if there are Python syntax errors or undefined names
+        ruff check rq tests --show-source --exit-zero
```

Comment:
I should update the comment above this line, but I also think we want don't want `--exit-zero`. We want to see the failure if there are syntax or other errors. I'll update this after other feedback.